OWML.ModHelper.Menus/ModInputCombinationPopup.cs

```diff
@@ -0,0 +1,185 @@
+﻿using OWML.Common;
+using OWML.Common.Menus;
+using OWML.ModHelper.Input;
+using OWML.ModHelper.Events;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using System;
+using System.Collections.ObjectModel;
+using System.Reflection;
+using System.Linq;
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModInputCombinationPopup : PopupMenu
+    {
+        public ILayoutManager Layout { get; private set; }
+        public event Action OnPopupReset;
+        public ReadOnlyCollection<KeyCode> KeyCodes => _combination.AsReadOnly();
+        public string Combination => string.Join("+", _combination.Select(ModInputLibrary.KeyCodeToString).ToArray());
+
+        private SubmitAction _resetAction;
+        private ButtonWithHotkeyImageElement _resetButton;
+        private List<KeyCode> _combination = new List<KeyCode>();
+        private IModInputHandler _inputHandler;
+        private bool _wasReleased = true;
+        protected SingleAxisCommand _resetCommand;
+
+        protected override void InitializeMenu()
+        {
+            base.InitializeMenu();
+            if (_resetAction != null)
+            {
+                _resetAction.OnSubmitAction += this.InvokeReset;
+            }
+        }
+
+        public override void Deactivate()
+        {
+            var component = _resetAction.GetComponent<UIStyleApplier>();
+            component?.ChangeState(UIElementState.NORMAL, true);
+            base.Deactivate();
+        }
+
+        protected virtual void InvokeReset()
+        {
+            _combination.Clear();
+            UpdateContents();
+        }
+
+        protected override void Update()
```

Comment:
Can you extract the contents into methods? It's kinda long

Reply:
done